src/services/resource-remover.js

```diff
@@ -3,7 +3,7 @@ const P = require('bluebird');
 function ResourceRemover(Model, params) {
   this.perform = () =>
     new P((resolve, reject) => {
-      Model.remove({ _id: params.recordId }, (err) => {
```

Comment:
Do you know the minimal Mongoose version where the new `deleteOne` and `deleteMany` functions where introduced.

We need to ensure that we won't break some existing projects.
Or if we do (break things), announce a breaking change (ie major version) with the known Mongoose versions this new agent won't support anymore.

Reply:
@arnaudbesnier these function exist since [version 4.9.0](https://github.com/Automattic/mongoose/blob/master/History.md#490--2017-03-13)

They are deprecated in the version 5, but I cannot find the exact version number for this deprecation.

I suppose that we'll have to add a `peerDependencies` entry in the `package.json` file of this project, to declare the minimal needed version for mongoose. It'll make the things clearer about the expectations of this package.

Reply:
ok, technically speaking, it is a breaking change for projects using mongoose versions < 4.9.
It should reflect in the commit message.

Then before the release we should create an "upgrade to v7" note like this:
https://docs.forestadmin.com/documentation/how-tos/maintain/upgrade-to-v6
That, at least, explains what v7 breaks.

This note would be restricted to the Forest Express Mongoose agent (which something we have never done yet).

(As lumber generate projects using a Mongoose version `~5.8.2`, no problem for newly generated lumber projects.)

Reply:
Maybe we could consider releasing without a breaking change (since it has a cost for us (docs + maintenance) and clients (following many upgrade notes)). Something like (ugly-pseudo-code coming):

```js
return IslandModel[ IslandModel.deleteMany ? deleteMany : remove ]({}); // sorry.
```

Just a suggestion though, I'm not even sure myself if it's a good move!

Reply:
@arnaudbesnier and @rap2hpoutre `deleteMany` is already used on `master` in 6 places.

So I suppose that the current code is already not compatible with mongoose < 4.9

In that case, I suppose that the change is not a breaking change in this case.